azerothcore/azerothcore-wotlk
# fix(Core/DB): Hardcoded gossip texts

## Removing hardcoded texts
## CHANGES PROPOSED:
- Migrating Lady Pestor hardcoded string gossip texts to the database
- New hooks that allow getting string texts based on locale
- Increased MenuID size for gossip_menu

The gossip strings were added to the acore_strings because adding them to the gossip_menu_options or npc_text didn't fit the logic behind this. Because we needed an easy way to store text, storing them on npc_text would be a waste and confusing since we only need the string and the rest is wasted resources. Also we needed the locales to change the language of these strings and more complex implementation would be necessary. The same goes for gossip_menu_option table. Too many conflicts with menuIds since they are primary keys and would be really hard to force the gossip text on these tables.

With acore_strings we have all we need. A default text for enUS and other locale texts that can be used if needed. All I did was read the locale and provide the string acordingly.

I'm gonna try to fix as many as time allows me to on the mentioned issue however a wiki entry on how to avoid hardcoded texts should be written.

I'm open to suggestions if you have a better way to engineer this since I only went for a fast solution that works and I think that something is better than nothing!

**Edit:** I forgot to add my two cents on something else.
I was thinking about adding a few columns to gossip_menu_options, for example bitmasks, to define conditions on the gossips. In this example, for Lady Pestor, the player is required to have a quest incomplete so the player gets the gossip option to complete the quest.
If we had a column that flags that a quest must be in incomplete state to display the gossip_menu_option, we could help devs script stuff faster and without core changes. I still like to be able to give diferent actions to gossips because it gives core scripts a lot of power.

## ISSUES ADDRESSED:
- Closes part of #2444



## TESTS PERFORMED:
- compiled on windows 10
- tested in-game




## HOW TO TEST THE CHANGES:

.npc add 1749
.npc delete (Clean up the added npc)

Since I don't have a diferent locale on my client only the default language will be used on my test (enUS). However I added the possibility to integrate other languages. If you wish to test them, on acore_strings check for ids 2697, 2698, 2699 and 2700 and add your traduced text.

## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
